tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_evaluate_diag_metric_tensor(self, tol):
+        """Test that a diagonal metric tensor evaluates correctly"""
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate metric tensor
+        g = circuit.metric_tensor(a, b, c)
+
+        # check that the metric tensor is correct
+        expected = np.array([1, np.cos(a)**2, (3-2*np.cos(a)**2*np.cos(2*b)-np.cos(2*a))/4])/4
+        assert np.allclose(g, np.diag(expected), atol=tol, rtol=0)
+
+    def test_evaluate_block_diag_metric_tensor(self, tol):
+        """Test that a block diagonal metric tensor evaluates correctly,
+        by comparing it to a known analytic result as well as numerical
+        computation."""
+        dev = qml.device('default.qubit', wires=3)
+
+        def non_parametrized_layer(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.RX(c, wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            qml.RZ(a, wires=0)
+            qml.Hadamard(wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(b, wires=1)
+            qml.Hadamard(wires=0)
+
+        a = 0.5
+        b = 0.1
+        c = 0.5
+
+        @qml.qnode(dev)
+        def final(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            qml.RY(f, wires=1)
+            qml.RZ(g, wires=2)
+            qml.RX(h, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        params = [-0.282203, 0.145554, 0.331624, -0.163907, 0.57662, 0.081272]
+        x, y, z, h, g, f = params
+
+        G = final.metric_tensor(x, y, z, h, g, f)
+
+        # ============================================
+        # Test block diag metric tensor of first layer is correct.
+        # We do this by comparing against the known analytic result.
+        # First layer include the non_parametrized_layer,
+        # followed by observables corresponding to generators of:
+        #   qml.RX(x, wires=0)
+        #   qml.RY(y, wires=1)
+        #   qml.RZ(z, wires=2)
+
+        G1 = np.zeros([3, 3])
+
+        # diag elements
+        G1[0, 0] = np.sin(a)**2/4
+        G1[1, 1] = (
+            16 * np.cos(a) ** 2 * np.sin(b) ** 3 * np.cos(b) * np.sin(2 * c)
+            + np.cos(2 * b) * (2 - 8 * np.cos(a) ** 2 * np.sin(b) ** 2 * np.cos(2 * c))
+            + np.cos(2 * (a - b))
+            + np.cos(2 * (a + b))
+            - 2 * np.cos(2 * a)
+            + 14
+        ) / 64
```

Comment:
how come some expressions look linted, but others don't?

Reply:
Any large mathematical expressions I blacked manually in a 'scratch' python file so that they would be easy to read.

I avoided blacking the entire file, as it has not been blacked before, and legacy code would show up in the diff. It would also cause merge conflicts with existing open PRs.

Prior to release/once we have merged (most) open PRs, we should have a PR that simply blacks the entire repo.